### ISSUE TITLE
ruff check does not warn about duplicate if TYPE_CHECKING: import clauses

### ISSUE THREAD

**Issue by ktns** [astral-sh/ruff#20690]
### Summary

Hi.

When I was poking around GitHub Copilot Agent, it emitted code like below and `ruff check` does not warn anything.

```python
if TYPE_CHECKING:
    from collections.abc import Iterable


if TYPE_CHECKING:
    from ase import Atoms
```

I think superfluous `if TYPE_CHECKING:` should be warned. What do you think? 

### Version

0.11.11

**Comment by njhearp** [comment]
If this is implemented I would suggest expanding it to another case. The same logic could also apply to duplicate `if __debug__`. Perhaps this could apply to version/platform (`sys.version_info >= (3, 13)`, etc.) checking as well.

**Comment by maltevesper** [comment]
The general case would be

```
if static_condition:
   dosomething()

if static_condition:
   dosomething2()
```

but I think it would be tricky to infer that `dosomething()` does not change `static_condition`.
Personally there are other rules which are higher priority to me, but it sounds like a nice addition.

**Comment by ktns** [comment]
I think defining a general rule that covers all static conditions is too difficult.

My original rationale was that ruff already has TC00N which generates/modifies `if TYPE_CHECKING` clause when fixing, so it's already a special case.

**Comment by njhearp** [comment]
Yes, I agree the cases I mentioned would not make sense.
